api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java

```diff
@@ -828,20 +828,22 @@ public HL7InArchive getHL7InArchiveByUuid(String uuid) throws APIException {
 	public Message processHL7Message(Message message) throws HL7Exception {
 		// Any post-parsing (pre-routing) processing would go here
 		// or a module can use AOP to do the post-parsing
-		
-		Message response;
+
+		Transportable response;
 		try {
-			if (!router.canProcess(message)) {
+			/*i*//*f (!router.canProcess(message)) {
 				throw new HL7Exception("No route for hl7 message: " + message.getName()
 				        + ". Make sure you have a module installed that registers a hl7handler for this type");
-			}
-			response = router.processMessage(message);
+			}*/
+			
+			 response = router.processMessage((Transportable) message);
+			//response = router.processMessage(message);
 		}
-		catch (ApplicationException e) {
+		catch (Exception e) {
```

Comment:
```suggestion
		catch (HL7Exception e) {
```

`HL7Exception` is the upper bound of exceptions thrown by the HAPI library.